pkg/ingress/controller/apisix_route.go

```diff
@@ -211,18 +212,15 @@ func (c *ApisixRouteController) sync(rqo *RouteQueueObj) error {
 		log.Errorf("invalid resource key: %s", key)
 		return fmt.Errorf("invalid resource key: %s", key)
 	}
-
-	apisixIngressRoute, err := c.apisixRouteList.ApisixRoutes(namespace).Get(name)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			log.Infof("apisixRoute %s is removed", key)
-			return nil
-		}
-		runtime.HandleError(fmt.Errorf("failed to list apisixRoute %s/%s", key, err.Error()))
-		return err
-	}
 	switch {
 	case rqo.Ope == UPDATE:
+		apisixIngressRoute, err := c.apisixRouteList.ApisixRoutes(namespace).Get(name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.Errorf("apisixRoute %s is removed", key)
```

Comment:
If the apisixRoute is removed, we don't need to retry, so just return `nil`.

Reply:
Yes